# Subject: Classical Data Analysis

## Session 1 - Regression

### Exercise 2 


Implementation of Python StatsModels package with Quandl integration to calculate the Linear regression with one variable.  

Considering the OLS presented in Demo 3 (Outbound tourism statistics for Spain), develop a new regression analysis based on the Quandl dataset “Inbound tourism statistics for Spain”. This dataset is available at https://www.quandl.com/data/UTOR/INTUR_ESP

Use the field “Tourist arrivals at national borders – Thousands” as independent variable and the field “Travel - US$ Mn” as dependent variable.

- Interpret and discuss the OLS Regression Results. 
- Commit scripts in your GitHub account. You should export your solution code (.ipynb notebook) and push it to your repository “ClassicalDataAnalysis”.

The following are the tasks that should complete and synchronize with your repository “ClassicalDataAnalysis” until October 13. Please notice that none of these tasks is graded, however it’s important that you correctly understand and complete them in order to be sure that you won’t have problems with further assignments.


## Import Quandl data using your API Key

In [2]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import quandl

quandl.ApiConfig.api_key = 'xVbFxivBYN4AK2jH3U6N'

In [3]:
data = quandl.get('UTOR/INTUR_ESP', authtoken='xVbFxivBYN4AK2jH3U6N')

In [4]:
data.head()

Tourist arrivals at national borders - Thousands  \
Date                                                           
1995-12-31                                           34920.0   
1996-12-31                                           36221.0   
1997-12-31                                           39553.0   
1998-12-31                                           41892.0   
1999-12-31                                           45440.0   

            Tourism expenditure in the country - US$ Mn  Travel - US$ Mn  \
Date                                                                       
1995-12-31                                      27369.0          25368.0   
1996-12-31                                      29751.0          27168.0   
1997-12-31                                      28649.0          26185.0   
1998-12-31                                      31592.0          29117.0   
1999-12-31                                      33784.0          31214.0   

            Passenger transport - US$ Mn  
Date                                      
1995-12-31                        2001.0  
1996-12-31                        2583.0  
1997-12-31                        2464.0  
1998-12-31                        2475.0  
1999-12-31                        2570.0

## Regression model with Statsmodels and without a constant

In [6]:
X = data['Tourist arrivals at national borders - Thousands'].values
y = data['Travel - US$ Mn'].values

In [9]:
model = sm.OLS(y, X).fit()

In [10]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.967
Model:                            OLS   Adj. R-squared:                  0.965
Method:                 Least Squares   F-statistic:                     551.8
Date:                Sun, 08 Oct 2017   Prob (F-statistic):           1.68e-15
Time:                        17:54:18   Log-Likelihood:                -209.65
No. Observations:                  20   AIC:                             421.3
Df Residuals:                      19   BIC:                             422.3
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
x1             0.9010      0.038     23.491      0.000       0.821       0.981
==============================================================================
Omnibus:                        6.516   Durbin-Watson:                   0.171
Prob(Omnibus):                  0.038   Jarque-Bera (JB):                1.723
Skew:                           0.056   Prob(JB):                        0.422
Kurtosis:                       1.566   Cond. No.                         1.00
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

### Interpreting the Table 

Our R-squared is 0.967, that means the model is pretty good at predicting the dependent variable with the arrivals of tourists.   
The coeficient is 0.9010, when the value of the independent variable changes by 1 the predicted value of the dependent variable changes by 0.9010 in the same direction.   
For the hypothesis testing with the P-value we have to reject the null hypothesis, and probably accept the validity of this model.

## Regression model with Statsmodels and with a constant

In [12]:
X = sm.add_constant(X)

In [13]:
model = sm.OLS(y, X).fit()

In [14]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.814
Model:                            OLS   Adj. R-squared:                  0.803
Method:                 Least Squares   F-statistic:                     78.61
Date:                Sun, 08 Oct 2017   Prob (F-statistic):           5.50e-08
Time:                        18:00:10   Log-Likelihood:                -203.23
No. Observations:                  20   AIC:                             410.5
Df Residuals:                      18   BIC:                             412.5
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -3.832e+04   9518.711     -4.026      0.001   -5.83e+04   -1.83e+04
x1             1.6339      0.184      8.866      0.000       1.247       2.021
==============================================================================
Omnibus:                        1.016   Durbin-Watson:                   0.368
Prob(Omnibus):                  0.602   Jarque-Bera (JB):                0.788
Skew:                          -0.130   Prob(JB):                        0.674
Kurtosis:                       2.063   Cond. No.                     3.33e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 3.33e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

### Interpreting the Table 

Our R-squared is now lower than before, the model isnt as good ad the previous at predicting the value of the dependent variable   
The coeficient is 1.6339, when the value of the independent variable changes by 1 the predicted value of the dependent variable changes by 1.6339 in the same direction.   
the coeficient for the constant is the value of the dependent variable when the independent variable is zero.   
For the hypothesis testing with the P-value we have to reject the null hypothesis, and probably accept the validity of this model.